File: src/main/java/com/amarpreet/logparser/data/LogData.java
package com.amarpreet.logparser.data;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Value;

import java.util.Date;

@Value
@Builder
@AllArgsConstructor
public class LogData {
    private String ipAddress;
    private Date ldate;
    private String url;
    private Integer httpStatus;
}

File: src/models/LogData.js
/**
 * LogData model - represents a single log entry
 */
class LogData {
  constructor(ipAddress, date, url, httpStatus) {
    this.ipAddress = ipAddress;
    this.date = date;
    this.url = url;
    this.httpStatus = httpStatus;
  }
}

module.exports = LogData;
